lib/instrumentation/span.js

```diff
@@ -42,6 +43,8 @@ function Span (transaction, name, type, opts) {
     this._recordStackTrace()
   }
 
+  this.transaction._captureBreakdown(this)
```

Comment:
What's the rationale behind moving this here? IIANM, this will cause spans to be included in the breakdown if they're started before but end after the transaction ends, and they would have a "self_time.sum.us" of zero. Spans should not be reported if they start or end after the transaction.

Reply:
The last 3 acceptance tests in the breakdown doc should cover this.

Reply:
It needs to be captured at the start to be able to properly capture the self-time it should subtract from the parent.

Reply:
This is what I just did locally to test the second last acceptance test. Note that I had to move the _captureBreakdown call back to span end, otherwise it fails.

```diff
diff --git a/lib/instrumentation/span.js b/lib/instrumentation/span.js
index cf9a8ce..6e532b4 100644
--- a/lib/instrumentation/span.js
+++ b/lib/instrumentation/span.js
@@ -43,8 +43,6 @@ function Span (transaction, name, type, opts) {
     this._recordStackTrace()
   }
 
-  this.transaction._captureBreakdown(this)
-
   this._agent.logger.debug('start span %o', { span: this.id, parent: this.parentId, trace: this.traceId, name: name, type: type })
 }
 
@@ -62,6 +60,7 @@ Span.prototype.end = function (endTime) {
   this._timer.end(endTime)
   this._agent._instrumentation._recoverTransaction(this.transaction)
 
+  this.transaction._captureBreakdown(this)
   this.ended = true
   this._agent.logger.debug('ended span %o', { span: this.id, parent: this.parentId, trace: this.traceId, name: this.name, type: this.type })
   this._agent._instrumentation.addEndedSpan(this)
diff --git a/test/metrics/breakdown.js b/test/metrics/breakdown.js
index 1e389a6..1e8e094 100644
--- a/test/metrics/breakdown.js
+++ b/test/metrics/breakdown.js
@@ -460,6 +460,36 @@ test('acceptance', t => {
     })
   })
 
+  t.test('with span ending after transaction', t => {
+    resetAgent(8, conf, ({ metricsets }) => {
+      const found = {
+        transaction: finders.transaction(metricsets),
+        transaction_span: finders['transaction span'](metricsets),
+      }
+
+      t.ok(found.transaction, 'found a transaction metricset')
+      t.deepEqual(found.transaction.samples, {
+        'transaction.duration.count': { value: 1 },
+        'transaction.duration.sum.us': { value: 20 },
+        'transaction.breakdown.count': { value: 1 }
+      }, 'sample values match')
+
+      t.ok(found.transaction_span, 'found a transaction span metricset')
+      t.deepEqual(found.transaction_span.samples, {
+        'span.self_time.count': { value: 1 },
+        'span.self_time.sum.us': { value: 10 }
+      }, 'sample values match')
+
+      agent._metrics.stop()
+      t.end()
+    })
+
+    var transaction = agent.startTransaction('foo', 'bar', { startTime: 0 })
+    var span = agent.startSpan('s name', 's type', { startTime: 10 })
+    transaction.end(null, 20)
+    if (span) span.end(30)
+  })
+
```

Reply:
All the acceptance tests are now implemented, with some improvements to the existing ones to more closely conform to what is in the design doc.